### PR TITLE
[env/github] fix auditorium v1 glow + badges location hiding

### PR DESCRIPTION
**Pull request by whoaitsaimz**
Hopefully actually closes https://github.com/sparkletown/internal-sparkle-issues/issues/817

~~Added !important to all overrides. A horrible way to do this.~~  Updated by malaika w / specifishity ;-) 

Riff off of https://github.com/sparkletown/sparkle/pull/1753

### CODE REVIEW

**Comment by 0xdevalias** [comment]
@whoaitsaimz Have you actually tested this change this time? I'd rather not deploy another PR for this without it being actually tested first.. this has been a source of much distraction today.

**Comment by malaikaaiyar** [comment]
I tested it out and if you sit in a section there is still a glow

Screen Shot 2021-07-08 at 10 14 41 PM

In the sections view, it looks like this:
Screen Shot 2021-07-08 at 10 15 19 PM

**Comment by whoaitsaimz** [comment]
The problem is my customOverrides.scss file is not taking priority. @malaikaaiyar would love your help w/ completing this one

My "hide time on venue & venues visited" no longer works either which makes me think there may have been a change to the code / loading order or something else that reduced priority of this file

**Comment by whoaitsaimz** [comment]
Glenn linked us to https://specifishity.com/ and it's a great reference.

However, when specificity is the same, order of loading DOES matter.

Ref: https://css-tricks.com/precedence-css-order-css-matters/

So I suggest, again, that we check where & how customOverrides.scss is being compiled relative to Auditorium.scss

This way we don't have to use !important flags -- the "nuclear" option, on every line, which is just silly.

If that can't be done this weekend, then please verify and merge existing PR so we can get this quickly finalized.

**Comment by malaikaaiyar** [comment]
Okay. I will take this on and complete it today. Thanks for the start though!

**Comment by malaikaaiyar** [comment]
This is ready for review. I've gotten rid of all the !important and made the css selectors specific enough to take priority

**Comment by 0xdevalias** [comment]
> However, when specificity is the same, order of loading DOES matter.

> So I suggest, again, that we check where & how customOverrides.scss is being compiled relative to Auditorium.scss

@whoaitsaimz As I have told you elsewhere (but adding it here for completeness); it's correct that order has an impact, but relying on order is a brittle antipattern that doesn't work in modern web apps that use bundled assets. It's unstable, unreliable, and shouldn't be used. Specificity is the only true way to resolve this, which is why I linked you to those resources.

File: src/components/templates/Auditorium/components/Section/Section.scss [hunk 1]
```diff
@@ -48,8 +48,6 @@ $spacing: calc(1 * min(var(--section-seat-size), var(--section-seat-size-min)));
     width: 100%;
     height: 100%;
     border: thin $black;
-    border-radius: $border-radius--lg;
-    box-shadow: $iframe-box-shadow;
```
Comment (0xdevalias) [inline src/components/templates/Auditorium/components/Section/Section.scss:52]: Why were these removed from here? I believe the intent was for all the changes to be made within `src/scss/customOverrides.scss` was it not?

File: src/scss/customOverrides.scss [hunk 2]
```diff
@@ -5,23 +5,23 @@
...
 .Badges {
   &__visits {
-    display: none;
+    display: none !important;
   }
   &__visit {
-    display: none;
+    display: none !important;
   }
 }
```
Comment (0xdevalias) [inline src/scss/customOverrides.scss:13]: These still seem to use `!important`, which goes against what you suggested to be the case in https://github.com/sparkletown/sparkle/pull/1755#issuecomment-877652052 ? Whys that?

**Review by 0xdevalias — CHANGES_REQUESTED** [review]
Questions/comments.

File: src/components/templates/Auditorium/components/Section/Section.scss (hunk 1 above)
Comment (malaikaaiyar) [reply src/components/templates/Auditorium/components/Section/Section.scss:52]: Oh I thought this was just for auditorium v1 changes. I'll put that in customOverrides then

File: src/scss/customOverrides.scss (hunk 2 above)
Comment (malaikaaiyar) [reply src/scss/customOverrides.scss]: fixed in d414cb523b6cf736e1693468c5beeee0686bb8b4

File: src/components/templates/Auditorium/components/Section/Section.scss (hunk 1 above)
Comment (malaikaaiyar) [reply src/components/templates/Auditorium/components/Section/Section.scss:52]: Those styles were removed from `AllSectionPreviews.scss` which is the other auditorium v2 component, so I thought that I should follow that. I'll add both back to their files and then 'remove' them in customOverrides

File: src/scss/customOverrides.scss (hunk 2 above)
Comment (malaikaaiyar) [reply src/scss/customOverrides.scss]: fixed in c129a927c7a43aec64d1228a2bf25f6194528f40

File: src/components/templates/Auditorium/components/Section/Section.scss (hunk 1 above)
Comment (malaikaaiyar) [reply src/components/templates/Auditorium/components/Section/Section.scss]: added all overriding styles to customOverrides in c129a927c7a43aec64d1228a2bf25f6194528f40

File: src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss [hunk 3]
```diff
@@ -110,7 +110,10 @@ $all-section-previews-bottom-spacing: 50px;
   }
 
   &__iframe {
-    height: 100%;
     width: 100%;
+    height: 100%;
+    border: thin $black;
+    border-radius: $border-radius--lg;
+    box-shadow: $iframe-box-shadow;
   }
```
Comment (0xdevalias) [inline src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss:118]: Hrmm.. still not sure this is right? The only file I expect to see any changes from on this PR should be in `src/scss/customOverrides.scss`?

File: src/components/templates/Auditorium/components/Section/Section.scss [hunk 4]
```diff
@@ -7,8 +7,6 @@ $section-seat-size: $avatarSize;
 $section-seat-size-min: $avatarSizeMin;
 $section-seat-spacing: 3px;
 
-$iframe-box-shadow: 2px 2px $spacing--lg $white;
-
```
Comment (0xdevalias) [inline src/components/templates/Auditorium/components/Section/Section.scss:11]: Hrmm.. still not sure this is right? The only file I expect to see any changes from on this PR should be in `src/scss/customOverrides.scss`?

File: src/scss/constants.scss [hunk 5]
```diff
@@ -90,6 +90,8 @@ $spacing--lg: 16px;
 $spacing--xl: 24px;
 $spacing--xxl: 30px;
 
+$iframe-box-shadow: 2px 2px $spacing--lg $white;
+
```
Comment (0xdevalias) [inline src/scss/constants.scss:94]: Hrmm.. still not sure this is right? The only file I expect to see any changes from on this PR should be in `src/scss/customOverrides.scss`?

**Review by 0xdevalias — CHANGES_REQUESTED** [review]
Looks like a few more unrelated changes sneaking in here.

File: src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss (hunk 3 above)
Comment (malaikaaiyar) [reply src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss:118]: @0xdevalias, the changes are not unrelated. In a past pr, someone (i think @whoaitsaimz) deleted the styles from auditorium v2 so that it doesn't have the glow. I re-added the glow styles to the components (needed to share the iframe-box-shadow through constants.scss) and "re-removed" it through customOverrides.scss.

Comment (0xdevalias) [reply src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss:118]: Are you sure they just weren't removed from the code in general? (eg. by @mike-lvov)

Searching for `border-radius: $border-radius--lg;` I don't see it in this file on `staging` either.

Comment (0xdevalias) [reply src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss:118]: https://github.com/sparkletown/sparkle/blob/staging/src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss#L108-L115

And according to the history:

- https://github.com/sparkletown/sparkle/commits/staging/src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss

This file hasn't been touched in `staging` since it was merged by @mike-lvov in

- https://github.com/sparkletown/sparkle/pull/1251

Comment (malaikaaiyar) [reply src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss:118]: You're right. I'll make sure this only touches v1 stuff and `customOverrides.scss`

Comment (malaikaaiyar) [reply src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss]: I basically just reverted the last commit

Comment (malaikaaiyar) [reply src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss]: fixed in c78545c

Comment (whoaitsaimz) [reply src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.scss]: Correct, I did not change ANYTHING in any file other than customOverrides.scss ;-)

**Review by sunny-viktoryia — APPROVED** [review]
Approving to unblock the PR. :)
@alexey-petrov has tested and confirmed that it works.

**Comment by alexey-petrov** [comment]
rebased & tested
seems to be working properly
image

**Comment by whoaitsaimz** [comment]
Thank you!!!
